src/types/index.ts

```diff
@@ -171,8 +171,14 @@ export type Widget = {
   _id: ObjectId
   title: string
   type: WidgetType
+  // Added this because the bookmarks field is required on the Collection interface below, but
+  // previously did not exist on the Widget type, which was causing a type error when trying to
+  // map over the user's MySpace to render the widgets.
+  bookmarks?: MongoBookmark[]
   // default indicates if the widget is automatically added for the user
   default?: boolean
+  // id is for dnd-kit
+  id?: string
```

Comment:
As the comment says, the difference in the presence of the `bookmarks` field was causing issues in a union type down the line. I believe we discussed this a few months back and a potential refactor of the MySpace-related types, but can't remember the decision. I added both `bookmarks` and `id` as optional fields, but let me know if that is potentially problematic and I'd be happy to take another look/create a new type.

Reply:
Is there a way we could use `_id` as the id field for dnd-kit? Or does it have to be `id`?